Formula/elasticsearch-full.rb

```diff
@@ -1,21 +1,30 @@
 class ElasticsearchFull < Formula
   desc "Distributed search & analytics engine"
   homepage "https://www.elastic.co/products/elasticsearch"
-  url "https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-7.6.0-darwin-x86_64.tar.gz?tap=elastic/homebrew-tap"
+  if OS.mac?
+    url "https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-7.6.0-darwin-x86_64.tar.gz?tap=elastic/homebrew-tap"
+    sha256 "eef8661778601a45a93824439c352116502bc92cfc751da55fec3d8459dc58d6"
+  else
+    url "https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-7.6.0-no-jdk-linux-x86_64.tar.gz?tap=elastic/homebrew-tap"
+    sha256 "19f0aa712ad1e5a4b45abfee11ea0edb3f64c56e53c1c6e50b1cecb1bd847886"
+  end
   version "7.6.0"
-  sha256 "eef8661778601a45a93824439c352116502bc92cfc751da55fec3d8459dc58d6"
-  conflicts_with "elasticsearch"
-  conflicts_with "elasticsearch-oss"
 
   bottle :unneeded
 
+  depends_on :java => "1.8" unless OS.mac?
+
+  conflicts_with "elasticsearch"
+  conflicts_with "elasticsearch-oss"
+
   def cluster_name
     "elasticsearch_#{ENV["USER"]}"
   end
 
   def install
     # Install everything else into package directory
-    libexec.install "bin", "config", "jdk.app", "lib", "modules"
+    libexec.install "bin", "config", "lib", "modules"
+    libexec.install "jdk.app" if OS.mac?
```

Comment:
This will need a different conditional after switching to the distribution that bundles a JDK.